packages/style-engine/class-wp-style-engine-css-rules-container.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Rules_Container
+ *
+ * A container for WP_Style_Engine_CSS_Rule objects.
+ *
+ * @package Gutenberg
+ */
+
+if ( ! class_exists( 'WP_Style_Engine_CSS_Rules_Container' ) ) {
```

Comment:
New class with a similar interface to `WP_Style_Engine_CSS_Rule`

This is intentional so that the methods can be called where there is a collection of rules + nested rules, e.g., in a single store.

Reply:
**TODO** - this class should extend `WP_Style_Engine_CSS_Rule`, or even look into adding these methods to `WP_Style_Engine_CSS_Rule` itself.

Reason being, with nested CSS, the container rule should also be able to have declarations.

This can be handled by the existing API, indicated by passing a container and not a selector:

```php
wp_style_engine_get_stylesheet_from_css_rules(
    array(
          'container' => 'fieldset',
          'declarations' => array( .... )
     ),
    array(
          'container' => 'fieldset',
           'container' => 'textarea',
          'declarations' => array( .... )
     )
)

/*
  Output:

   fieldset {
       ...declarations;
      textarea {
           ...declarations;
       }
    }
*/ 
```